baseledger/x/proof/client/rest/txutil.go

```diff
@@ -28,34 +25,62 @@ var (
 )
 
 func BuildClientCtx(clientCtx client.Context) (*client.Context, error) {
-	keyring, err := NewKeyringInstance()
-
-	keysList, err := keyring.List()
+	keyringInstance, err := NewKeyringInstance()
 	if err != nil {
-		logger.Errorf("error getting key list %v\n", err.Error())
-		return nil, errors.New("")
+		logger.Errorf("error getting keyring instance %v\n", err.Error())
+		return nil, err
 	}
 
-	if len(keysList) == 0 {
-		return nil, errors.New("")
+	// node can specify key address as env variable, otherwise it will use first from keylist
+	key, err := getKey(keyringInstance)
+	if err != nil {
+		logger.Errorf("error getting key %v\n", err.Error())
+		return nil, err
 	}
 
-	rand.Seed(time.Now().UnixNano())
-	min := 0
-	max := len(keysList) - 1
-	randomAccIdx := rand.Intn(max-min+1) + min
-
-	// every node should configure key for this purpose, and it should be first in key list
 	clientCtx = clientCtx.
```

Comment:
we added this randomized key for lakewood, but here i feel we should be more deterministic
i thought we can leave 2 options for nodes
1. specify key address using env variable
2. use first from keylist